app/routes/workspace/home.tsx

```diff
@@ -193,6 +200,137 @@ const ImportDialogContent = () => {
   );
 };
 
+const CreateNewProjectButton = () => {
+  const { t } = useTranslation();
+  const { open, onOpenChange } = useDialogHotkeys(['newProjectDialog']);
+  return (
+    <Dialog open={open} onOpenChange={onOpenChange}>
+      <CreateNewArtefactButton title={t('workspaces.newProject')} onClick={() => onOpenChange(true)} />
+      <DialogContent>
+        <NewProjectContent closeDialog={() => onOpenChange(false)} />
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+const NewProjectContent = ({ closeDialog }: { closeDialog: () => void }) => {
+  const { t } = useTranslation();
+  const [name, setName] = useState('');
+  const [defaultNamespace, setDefaultNamespace] = useState('');
+  const [groupId, setGroupId] = useState('');
+  const [projectId, setProjectId] = useState('');
+  const { artifactAlreadyExists, validateArtifactName, validateArtifactNamespace, validateProjectDetails } = useArtifactValidation();
+  const workspaces = useWorkspaces();
+  const { data } = useSortedProjects();
+  const { createProject } = useCreateProject();
+  const create = () => createProject({ defaultNamespace: defaultNamespace, name: name, groupId: groupId, projectId: projectId });
+
+  const nameValidation = useMemo(
+    () =>
+      workspaces.data?.find(w => w.name.toLowerCase() === name.toLowerCase()) ? artifactAlreadyExists(name) : validateArtifactName(name),
+    [artifactAlreadyExists, name, validateArtifactName, workspaces.data]
+  );
+  const defaultNamespaceValidation = useMemo(
+    () =>
+      data?.find(p => p.defaultNamespace.toLowerCase() === defaultNamespace.toLowerCase())
+        ? artifactAlreadyExists(defaultNamespace)
+        : validateArtifactNamespace(defaultNamespace),
+    [artifactAlreadyExists, data, defaultNamespace, validateArtifactNamespace]
+  );
+  const projectIdValidation = useMemo(
+    () =>
+      data?.find(p => p.artifactId.toLowerCase() === projectId.toLowerCase())
+        ? artifactAlreadyExists(projectId)
+        : validateProjectDetails(projectId, '-'),
+    [artifactAlreadyExists, data, projectId, validateProjectDetails]
+  );
+  const groupIdValidation = useMemo(
+    () =>
+      data?.find(p => p.groupId.toLowerCase() === groupId.toLowerCase())
+        ? artifactAlreadyExists(groupId)
+        : validateProjectDetails(groupId, '.'),
+    [artifactAlreadyExists, data, groupId, validateProjectDetails]
+  );
+
+  const hasErros = useMemo(() => nameValidation?.variant === 'error', [nameValidation]);
+
+  const createNewProject = () => {
+    console.log('createNewProject');
+    if (!hasErros) {
+      closeDialog();
+      create();
+    }
+  };
+
+  const enter = useHotkeys('Enter', createNewProject, { scopes: ['newProjectDialog'], enableOnFormTags: true });
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newName = e.target.value;
+    const updatedProjectId = changeHandler(name, newName, projectId, '-');
```

Comment:
I'm not sure if we should always update these other fields when the name changes... IMO we should only do this when these other fields are not yet modified manually...
We can easily check this e.g. by checking if the generation with the old name is equal to the current field name if yes, generate it with the new name.